lib/bolt/cli.rb

```diff
@@ -654,8 +654,21 @@ def generate_types
     # Initializes a specified directory as a Bolt project and installs any modules
     # specified by the user, along with their dependencies
     def initialize_project
-      project    = Pathname.new(File.expand_path(options[:object] || Dir.pwd))
-      config     = project + 'bolt.yaml'
+      name       = options[:object] || File.basename(Dir.pwd)
+      name_regex = /^[a-z][a-z0-9_]*$/
+      if name !~ name_regex
+        if options[:object]
+          raise Bolt::ValidationError, "The provided project name '#{name}' is invalid; "\
+            "project name must match #{name_regex.inspect}"
+        else
+          raise Bolt::ValidationError, "The current directory name '#{name}' will be used "\
+            "as the project name, and is invalid. Project name must match #{name_regex.inspect}"
```

Comment:
```suggestion
            "as the project name, and is invalid. Project name must begin with a lowercase "\
            "letter, and may only contain lowercase letters, digits, and underscores."
```

For users who aren't familiar with regex, this can be cryptic. I think adding a more verbose description of the requirements, similar to [Bolt projects documentation](https://puppet.com/docs/bolt/latest/projects.html#creating-a-bolt-project), would be helpful.

Reply:
"Must start with a letter and can include numbers, letters, or underscores". I cribbed this directly from the project class, so...I'll update there too.

Reply:
"Must not be able to moonlight as a strong password"